lib/ansible/module_utils/docker_common.py

```diff
@@ -381,7 +382,7 @@ def find_image(self, name, tag):
 
         self.log("Find image %s:%s" % (name, tag))
         images = self._image_lookup(name, tag)
-        if len(images) == 0:
+        if len(images) == 0 and name not in DOCKER_RESERVED_IMAGE_NAMES:
             # In API <= 1.20 seeing 'docker.io/<name>' as the name of images pulled from docker hub
```

Comment:
It took me a while to figure how 'name' could be in DOCKER_RESERVED_IMAGE_NAMES. And then it hit me, `docker_container`, and probably `docker_image`, split the `image` parameter on ':', and assume that the first element is 'name', and the second element is a 'tag'. 

I think it would be preferable to have the code recognize when it receives an ID or partial ID value, and search accordingly. It would be nice if this could work with partial IDs, but this code will not support that.